kornia/enhance/equalization.py

```diff
@@ -0,0 +1,344 @@
+    ih = torch.arange(2 * th - 1, -1, -1, device=interp_tiles.device).div(2 * th - 1)[None].T.expand(2 * th, tw)
+    ih = ih.unfold(0, th, th).unfold(1, tw, tw)  # 2 x 1 x TH x TW
+    iw = torch.arange(2 * tw - 1, -1, -1, device=interp_tiles.device).div(2 * tw - 1).expand(th, 2 * tw)
+    iw = iw.unfold(0, th, th).unfold(1, tw, tw)  # 1 x 2 x TH x TW
+
+    # compute row and column interpolation weigths
+    tiw = iw.expand((gw - 2) // 2, 2, th, tw).reshape(gw - 2, 1, th, tw).unsqueeze(0)  # 1 x GW-2 x 1 x TH x TW
+    tih = ih.repeat((gh - 2) // 2, 1, 1, 1).unsqueeze(1)  # GH-2 x 1 x 1 x TH x TW
+
+    # internal regions
+    tl, tr, bl, br = preinterp_tiles_equalized[:, 1:-1, 1:-1].unbind(3)
+    t = tiw * (tl - tr) + tr
+    b = tiw * (bl - br) + br
+    tiles_equalized[:, 1:-1, 1:-1] = tih * (t - b) + b
+
+    # corner regions
+    tiles_equalized[:, 0::gh - 1, 0::gw - 1] = preinterp_tiles_equalized[:, 0::gh - 1, 0::gw - 1, 0]
+
+    # border region (h)
+    t, b, _, _ = preinterp_tiles_equalized[:, 1:-1, 0].unbind(2)
+    tiles_equalized[:, 1:-1, 0] = tih.squeeze(1) * (t - b) + b
+    t, b, _, _ = preinterp_tiles_equalized[:, 1:-1, gh - 1].unbind(2)
+    tiles_equalized[:, 1:-1, gh - 1] = tih.squeeze(1) * (t - b) + b
+
+    # border region (w)
+    l, r, _, _ = preinterp_tiles_equalized[:, 0, 1:-1].unbind(2)
+    tiles_equalized[:, 0, 1:-1] = tiw * (l - r) + r
+    l, r, _, _ = preinterp_tiles_equalized[:, gw - 1, 1:-1].unbind(2)
+    tiles_equalized[:, gw - 1, 1:-1] = tiw * (l - r) + r
+
+    # same type as the input
+    return tiles_equalized.to(interp_tiles).div(255.)
+
+
+def equalize_clahe(input: torch.Tensor, clip_limit: float = 40., grid_size: Tuple[int, int] = (8, 8)) -> torch.Tensor:
+    r"""Apply clahe equalization on the input tensor.
+
+    Args:
+        input (torch.Tensor): images tensor to equalize with values in the range [0, 1] and shapes like
+                              :math:`(C, H, W)` or :math:`(B, C, H, W)`.
+        clip_limit (float): threshold value for contrast limiting. If 0 clipping is disabled. Default: 40.
+        grid_size (Tuple[int, int]): number of tiles to be cropped in each direction (GH, GW). Default: (8, 8).
+
+    Returns:
+        torch.Tensor: Equalized image or images with shape as the input.
+
+    Examples:
+        >>> img = torch.rand(1, 10, 20)
+        >>> res = equalize_clahe(img)
+        >>> res.shape
+        torch.Size([1, 10, 20])
+
+        >>> img = torch.rand(2, 3, 10, 20)
+        >>> res = equalize_clahe(img)
+        >>> res.shape
+        torch.Size([2, 3, 10, 20])
+
+    """
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input input type is not a torch.Tensor. Got {type(input)}")
+
+    if input.dim() not in [3, 4]:
+        raise ValueError(f"Invalid input shape, we expect CxHxW or BxCxHxW. Got: {input.shape}")
+
+    if input.numel() == 0:
+        raise ValueError("Invalid input tensor, it is empty.")
+
+    if input.max() > 1. or input.min() < 0.:
```

Comment:
I would remove this check as we dicussed in the augmentations might slow down processes. /cc @shijianjian